sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -364,9 +357,14 @@ function buildSqlParameter(value: RawSqlParameter): SqlParameter {
  * @param value
  */
 export function getRawSqlParameters(parameters: SqlParameter[] | undefined): any {
-  if (!Array.isArray(parameters)) {
+  if (parameters == undefined) {
     return undefined;
   }
+
+  if (!Array.isArray(parameters)) {
+    throw new TypeError(`SQL parameters must be an array of SqlParameter instances or undefined`);
```

Comment:
`getRawSqlParameters()` is called from 2 places:
- Building the sql filter from user input when they create/update a rule
- Building the sql action from user input when they create/update a rule

Ideally, I'd like to provide more context in the error message by moving of this check and throwing of the error to the caller of `getRawSqlParameters()`. That way, we can provide context on whether the problem was in the user given sql filter or the sql action.

Reply:
Since user input is now included in error message, this doesn't seem necessary?

Also, about the tests - typescript catches them at compile time and so it isn't possible to add them simply.

Reply:
Resolving this as in the latest version of the code, I dont see the error being thrown

Reply:
Including the user input in the error message definitely helps here. Resolving the comment thread